File: library/src/main/java/com/hm/iou/userinfo/business/VipStatusContract.java
package com.hm.iou.userinfo.business;

import com.hm.iou.base.mvp.BaseContract;

public interface VipStatusContract {

    interface View extends BaseContract.BaseView {

        void showCommUserInfoView();

        void showVipUserInfoView();

        /**
         * 显示头像
         *
         * @param url          头像地址
         * @param defIconResId 默认头像图片资源id
         */
        void showAvatar(String url, int defIconResId);

        void showDataLoading(boolean show);

        void showDataLoadError();

        /**
         * 显示VIP有效期
         *
         * @param validDate
         */
        void showVipValidDate(String validDate);
    }

    interface Presenter extends BaseContract.BasePresenter {

        void init();

        void getMemberInfo();

        void getPayInfo();
    }

}

File: library/src/main/java/com/hm/iou/userinfo/business/VipStatusContract.kt
package com.hm.iou.userinfo.business;

import com.hm.iou.base.mvp.BaseContract;
import com.hm.iou.userinfo.business.view.VipICouponItem
import com.hm.iou.userinfo.business.view.VipIHeaderModuleItem

interface VipStatusContract {

    interface View : BaseContract.BaseView {

        /**
         * 显示头部头像
         */
//        fun showHeaderInfo(headerUrl: String?, defaultAvatarResId: Int)

        /**
         * 显示普通用户信息
         */
        fun showNoVipUserInfoView(headerUrl: String?, defaultAvatarResId: Int, remindDay: Int?, listModule: List<VipIHeaderModuleItem>?, listCoupon: List<VipICouponItem>?)

        fun updateCouPonItem(position: Int)
        /**
         * 显示VIP用户信息
         *
         * @param vipValidDate
         */
        fun showVipUserInfoView(headerUrl: String?, defaultAvatarResId: Int, remindDay: Int?, vipValidDate: String?, listModule: List<VipIHeaderModuleItem>?, listCoupon: List<VipICouponItem>?)

    }

    interface Presenter : BaseContract.BasePresenter {

        fun init()

        fun getPayInfo()

        fun getCoupon(couponId: String, position: Int)

    }

}